src/timeseriesinsights/README.md

```diff
@@ -1,95 +1,129 @@
-Microsoft Azure CLI 'timeseriesinsights' Extension
-==========================================
+# Azure CLI timeseriesinsights Extension #
+This is the extension for timeseriesinsights
 
-This package is for the 'timeseriesinsights' extension:
-
-```sh
-az timeseriesinsights
+### How to use ###
+Install this extension using the below CLI command
 ```
-
-## Install
-
-```sh
 az extension add --name timeseriesinsights
 ```
 
-## Uninstall
-
-```sh
-az extension remove --name timeseriesinsights
+### Included Features ###
+#### timeseriesinsights environment ####
+##### Gen1 Create #####
 ```
-
-## Examples
-
-All commands are for Bash. For PowerShell, please set the variables like 
-
-```powershell
-$rg='rg1'
+az timeseriesinsights environment gen1 create --name "env1" --location westus --data-retention-time "P31D" \
+    --partition-key-properties name="DeviceId1" type="String" --sku name="S1" capacity=1 --resource-group "rg1"
 ```
 
-### Create a resource group for the environments
-
-```sh
-rg={resource_group_name}
-az group create --name $rg --location westus
+##### Gen1 Update #####
 ```
-
-### Create a standard environment
-
-```sh
-env={standard_environment_name}
-az timeseriesinsights environment standard create -g $rg --name $env --location westus --sku-name S1 --sku-capacity 1 --data-retention-time 31 --partition-key-properties DeviceId1 --storage-limit-exceeded-behavior PauseIngress
+az timeseriesinsights environment gen1 update --name "env1" --sku name="S1" capacity=2 --resource-group "rg1" \
+    --data-retention-time "P30D" --storage-limit-exceeded-behavior PurgeOldData
 ```
 
-### Create a storage account and use it to create a long-term environment
-
-```sh
-storage={storage_account_name}
-env_lt={longterm_environment_name}
-
-az storage account create -g $rg -n $storage --https-only
-key=$(az storage account keys list -g $rg -n $storage --query [0].value --output tsv)
-
-az timeseriesinsights environment longterm create -g $rg --name $env_lt --location westus --sku-name L1 --sku-capacity 1 --data-retention 7 --time-series-id-properties DeviceId1 --storage-account-name $storage --storage-management-key $key
+##### Gen2 Create #####
 ```
-
-### Create an event hub and use it to create an event source
-
-```sh
-ehns={eventhub_namespace}
-eh={eventhub_name}
-
-az eventhubs namespace create -g $rg -n $ehns
-es_resource_id=$(az eventhubs eventhub create -g $rg -n $eh --namespace-name $ehns --query id --output tsv)
-shared_access_key=$(az eventhubs namespace authorization-rule keys list -g $rg --namespace-name $ehns -n RootManageSharedAccessKey --query primaryKey --output tsv)
-
-az timeseriesinsights event-source eventhub create -g $rg --environment-name $env --name {es1} --key-name RootManageSharedAccessKey --shared-access-key $shared_access_key --event-source-resource-id $es_resource_id --consumer-group-name '$Default' --timestamp-property-name DeviceId
```

Comment:
I think it will be very helpful to include contextual examples like this one to demonstrate how to create relevant/supporting resources.